### ISSUE TITLE
OIDC refresh token is lost after ID token expiration

### ISSUE THREAD

**Issue by antoniomacri** [quarkusio/quarkus#11715]
**Describe the bug**
I've configured the OIDC extension for a web-app in order to refresh the tokens after expiration (refresh-expired=true).

By inspecting the q_session cookie I see it contains all the three tokens (id token, access token, refresh token), however the max-age is set as the access token expiration.

As a consequence the refresh token is never used, since just after the id/access token expiration the cookie is not even sent by the browser.

**Expected behavior**
The q_session cookie is sent after id token expiration and the refresh is performed consequently.

**Actual behavior**
The q_session expires with the id token and the server cannot perform refresh.

**Configuration**
```properties
quarkus.oidc.application-type=WEB_APP
quarkus.oidc.discovery-enabled=false
quarkus.oidc.auth-server-url=https://...
quarkus.oidc.authorization-path=/as/authorization.oauth2
quarkus.oidc.token-path=/as/token.oauth2
quarkus.oidc.jwks-path=/pf/JWKS
quarkus.oidc.client-id=Oauth_clientid
quarkus.oidc.credentials.secret=myclientsecret
quarkus.oidc.tls.verification=none
quarkus.oidc.authentication.redirect-path=/my/app
quarkus.oidc.token.refresh-expired=true
quarkus.oidc.authentication.cookie-path=${quarkus.http.root-path}
quarkus.oidc.authentication.verify-access-token=false
quarkus.oidc.roles.source=idtoken
quarkus.oidc.token.lifespan-grace=120
```

**Environment (please complete the following information):**
 - Quarkus 1.7.1

**Comment by sberyozkin** [comment]
@antoniomacri FYI, #11718

**Comment by antoniomacri** [comment]
Hi Sergey,

I've built the snapshot from master in order to test the fix, however it's still causing a redirect instead of silently refreshing the tokens. I'm using these properties (among others):

```
quarkus.oidc.token.lifespan-grace=120
quarkus.oidc.token.refresh-expired=true
quarkus.oidc.authentication.session-age-extension=8760h
```

I've tried to debug but I lost myself in the callbacks, tomorrow I'll try to investigate further... any suggestions on things to check are welcome.

Thanks again.
